File: 0739-daily-temperatures/0739-daily-temperatures.ts
function dailyTemperatures(temperatures: number[]): number[] {
    const stack: number[] = []
    const res = Array(temperatures.length).fill(0)
    for (let i=temperatures.length-1; i >= 0; i--) {
        while (stack.length && temperatures[stack.at(-1)] <= temperatures[i]) stack.pop()
        if (stack.length) res[i] = stack.at(-1) - i
        stack.push(i)
    }
    return res
};

File: 0739-daily-temperatures/0739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        n = len(temperatures)
        res, stack = [0] * n, []
        for i in range(n-1, -1, -1):
            while stack and temperatures[stack[-1]] <= temperatures[i]:
                stack.pop()
            res[i] = stack[-1] - i if stack else 0
            stack.append(i)
        return res